extensions/amp-analytics/0.1/test/test-vendors-new.js

```diff
@@ -0,0 +1,216 @@
+import {
+  ImagePixelVerifier,
+  mockWindowInterface,
+} from '../../../../testing/test-helper';
+import {Services} from '../../../../src/services';
+import {hasOwn} from '../../../../src/utils/object';
+import {macroTask} from '../../../../testing/yield';
+
+/* global require: false */
+const VENDOR_REQUESTS = require('./vendor-requests.json');
+
+describes.realWin(
+  'amp-analytics',
+  {
+    amp: {
+      extensions: ['amp-analytics'],
+    },
+    mockFetch: false,
+  },
+  function(env) {
+    let win, doc;
+    let requestVerifier;
+
+    beforeEach(() => {
+      win = env.win;
+      doc = win.document;
+      const wi = mockWindowInterface(env.sandbox);
+      wi.getLocation.returns(win.location);
+      requestVerifier = new ImagePixelVerifier(wi);
+    });
+
+    describe('Should not contain iframe transport if not whitelisted', () => {
+      for (const vendor in VENDOR_REQUESTS) {
+        it('test vendor: ' + vendor, () => {
+          const el = doc.createElement('amp-analytics');
+          el.setAttribute('type', vendor);
+          doc.body.appendChild(el);
+          const analyticsConfig = new AnalyticsConfig(el);
+          return analyticsConfig.loadConfig().then(config => {
+            if (
+              hasOwn(config, 'transport') &&
+              hasOwn(config.transport, 'iframe')
+            ) {
+              expect(config['transport']['iframe']).to.equal(
+                IFRAME_TRANSPORTS[vendor]
+              );
+            }
+          });
+        });
+      }
+    });
+
+    describe('vendor request tests', () => {
+      for (const vendor in VENDOR_REQUESTS) {
+        describe('analytics vendor: ' + vendor, function() {
+          let config;
+          let analytics;
+          beforeEach(done => {
+            const urlReplacements = Services.urlReplacementsForDoc(
+              doc.documentElement
+            );
+            window.sandbox
+              .stub(urlReplacements.getVariableSource(), 'get')
+              .callsFake(function(name) {
+                expect(this.replacements_).to.have.property(name);
+                const defaultValue = `_${name.toLowerCase()}_`;
+                return {
+                  sync: () => defaultValue,
+                };
+              });
+
+            window.sandbox
+              .stub(ExpansionOptions.prototype, 'getVar')
+              .callsFake(function(name) {
+                let val = this.vars[name];
+                if (val == null || val == '') {
+                  val = '!' + name;
+                }
+                return val;
+              });
+
+            analytics = getAnalyticsTag(doc, vendor);
+            analytics.layoutCallback().then(() => {
+              config = analytics.config_;
+              done();
+            });
+          });
+
+          it('test requests', function*() {
+            const outputConfig = {};
+            if (!config.requests) {
+              throw new Error(
```

Comment:
@estherkim Want to get your thoughts on this. We don't need to build the extension when running the test locally. However we do need to copy the vendor config json files to the dist/v0 folder to serve them. What is a good way include the command in the pre run? Thank you!

Reply:
Hmm the setup for unit tests would be a good place to run `vendorConfigs()`. It can always be run regardless of the `--nobuild` flag to make sure the json files are available.

before:
https://github.com/ampproject/amphtml/blob/ed9b51c6caa1824a79675e2992566b92333ffa0b/build-system/tasks/unit.js#L36-L45

suggested after:
```
  /** @override */
  async maybeBuild() {
    await vendorConfigs();

    if (!argv.nobuild) {
      await css();
      await compileJison();
    }
  }
```

Reply:
I see. This makes sense. Should be fine to add as `vendorConfigs()` runs pretty fast. I'll create a following PR to address this. Thank you Esther!

Reply:
cc @micajuine-ho